### PR TITLE
Implemented buildRoute() without using path module

### PR DESCRIPTION
**Pull request by rulych**
`buildRoute()` doesn't depend on the `path` NodeJS module, using instead regular expressions to get the same functionality.

### CODE REVIEW

File: src/domain/createBuildRoute/index.js [hunk 1]
```diff
@@ -43,4 +42,19 @@ function getParams(args) {
   return undefined;
 }
 
+function normalizeUrl(url) {
+  let result = url.replace(/[^:]\/\//g, '/'); // Gets rid of duplicated slashes (//)
```
Comment (notatestuser) [inline src/domain/createBuildRoute/index.js]:
```js
> normalizeUrl('/test//path')
'/tes/path'
```

Comment (notatestuser) [inline src/domain/createBuildRoute/index.js]: not sure if we care:
```js
> normalizeUrl('file:///test/path')
'file:/test/path'
```

Comment (notatestuser) [inline src/domain/createBuildRoute/index.js]: To fix the first one you might want to do
```js
url.replace(/[^:]\/\//g, (match) => match.replace('//','/'))
```
or
```js
url.replace(/([^:])\/\//g, (match, p1) => `${p1}/`)
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_function_as_a_parameter

File: src/domain/createBuildRoute/index.js [hunk 2]
```diff
@@ -43,4 +42,19 @@ function getParams(args) {
...
+  while (true) { // Interprets two dots (..), going up in the path for each occurrence
+    const newResult = result.replace(/[a-z0-9]*\/\.\.\/?/gi, '');
+    if (result === newResult) {
+      break;
+    }
+
+    result = newResult;
+  }
+  result = result.replace(/\.\//g, ''); // Gets rid of ./
```
Comment (notatestuser) [inline src/domain/createBuildRoute/index.js]: Would "./" only appear at the start?

Comment (rulych) [inline src/domain/createBuildRoute/index.js]: It wouldn't hurt trying to support it everywhere, I think.

File: src/domain/createBuildRoute/index.js [hunk 3]
```diff
@@ -42,10 +42,10 @@ function getParams(args) {
   return undefined;
 }
 
-function normalizeUrl(url) {
-  let result = url.replace(/[^:]\/\//g, '/'); // Gets rid of duplicated slashes (//)
+export function normalizeUrl(url) {
+  let result = url.replace(/(^|[a-z0-9])\/+/gi, '$1/'); // Gets rid of duplicated slashes (//)
```
Comment (notatestuser) [inline src/domain/createBuildRoute/index.js]: Being picky but `"base_//derp"` still won't be processed even though it's a valid path. How about `/(^|[\w-])\/+/gi`?

File: src/domain/createBuildRoute/index.js [hunk 4]
```diff
@@ -42,10 +42,10 @@ function getParams(args) {
...
   while (true) { // Interprets two dots (..), going up in the path for each occurrence
-    const newResult = result.replace(/[a-z0-9]*\/\.\.\/?/gi, '');
+    const newResult = result.replace(/((^\/)|[a-z0-9]*\/)\.\.\/?/gi, '$2');
```
Comment (notatestuser) [inline src/domain/createBuildRoute/index.js]: Same issue here? Use `\w` and include `-`?

File: src/domain/createBuildRoute/index.js (hunk 3 above)
Comment (rulych) [inline src/domain/createBuildRoute/index.js]: Fair enough.

**Pull request merged**